Support multiline \title tags in Rd files

jranke opened the pull request:
Fixes #178

hadley commented on R/package.r:
```diff
@@ -89,7 +89,9 @@ package_rd <- function(path) {
-    purrr::map_chr(c(1, 1))
+    unlist %>%
+    paste(collapse = " ") %>%
+    stringr::str_trim()
```
Please use `trimws` to avoid the stringr dep

hadley replied:
Also if you're going to take this approach you're going to need some unit tests to verify that it actually does what you expect.

jranke replied:
Ah, I did not know base::trimws exists. 

You are right, the approach fails e.g. when there is more than one \alias tag in the Rd file. So somehow we need to differentiate between lists with several elements originating from a multiline tag, (e.g. \title) and lists coming from several tags (e.g. several occurrences of \alias).

jranke commented:
I squashed the commits as they were partially just going back and forth. I also added some unit tests. You can see the result applied to one of my packages [here](http://jranke.github.io/mkin/reference/index.html).

jranke commented:
Sorry, I did not anticipate that squashing the commits would mess up the display of this review thread...

hadley commented on tests/testthat/test-extract_tag.R:
```diff
@@ -0,0 +1,31 @@
+This is a nice title starting with a newline
+}", rd)
+
+  rd_file(rd) %>%
```
I think you can simplify this by using `rd_text()`. Then you don't need to write to a temporary file

hadley commented on tests/testthat/test-extract_tag.R:
```diff
@@ -0,0 +1,31 @@
+    extract_tag("tag_title") %>%
+    expect_equal("This is a title\ncontaining a newline")
+})
+
```
I think it would also be useful to include a test of a title with inline Rd, e.g. `\title{\emph{emph} and \strong{strong}}`

JrEduardo commented:
This [modification](https://github.com/hadley/pkgdown/pull/212/commits/ac6a2528cb98933d2e79ac3f135a2ae756a1ea98#L183) in `build-reference.R` fails to treat other tags in the `\title{}`, for example `\title{\emph{emph} and \strong{strong}}`, as @hadley commented.

We need the `flatten_text` function. I think this works:

```
## Replace line 183 in build-reference.R by
out$title <- flatten_text(tags$tag_name[[1]])
```

Because now, the list's class is according with the functions `as_html.tag_*` ( on `rd-html.R` file). And as `flatten_text` uses `purrr::map_chr`, it should work.
